src/components/file/file.tsx

```diff
@@ -129,111 +118,42 @@ export class File {
-        this.handleFileChange = this.handleFileChange.bind(this);
-        this.handleChipSetChange = this.handleChipSetChange.bind(this);
-        this.handleFileDrop = this.handleFileDrop.bind(this);
-        this.handleKeyDown = this.handleKeyDown.bind(this);
-        this.handleKeyUp = this.handleKeyUp.bind(this);
-        this.handleChipInteract = this.handleChipInteract.bind(this);
-    }
-
-    public connectedCallback() {
-        this.initialize();
-    }
-
-    public componentDidLoad() {
-        this.fileInput = this.element.shadowRoot.getElementById(
-            this.fileInputId,
-        ) as HTMLInputElement;
-        this.chipSet = this.element.shadowRoot.querySelector(CHIP_SET_TAG_NAME);
-        this.initialize();
-    }
-
-    private initialize() {
-        if (!this.chipSet) {
-            return;
-        }
-
-        this.mdcTextField = new MDCTextField(
-            this.chipSet.shadowRoot.querySelector('.mdc-text-field'),
-        );
-    }
-
-    public disconnectedCallback() {
-        if (this.mdcTextField) {
-            this.mdcTextField.destroy();
-        }
-    }
-
     public render() {
-        return [
-            <input
-                hidden={true}
-                id={this.fileInputId}
-                onChange={this.handleFileChange}
-                type="file"
+        return (
+            <limel-file-dropzone
+                disabled={this.disabled || this.readonly || !!this.value}
                 accept={this.accept}
-                disabled={this.disabled || this.readonly}
-            />,
-            <limel-chip-set
-                class={{
-                    'is-file-picker': true,
-                    'shows-dropzone': true,
-                    'highlight-dropzone': this.isDraggingOverDropZone,
-                }}
-                disabled={this.disabled}
-                readonly={this.readonly}
-                invalid={this.invalid}
-                label={this.label}
-                leadingIcon="upload_to_cloud"
-                language={this.language}
-                onChange={this.handleChipSetChange}
-                onClick={this.handleClick}
-                onInteract={this.handleChipInteract}
-                onKeyDown={this.handleKeyDown}
-                onKeyUp={this.handleKeyUp}
-                required={this.required}
-                type="input"
-                value={this.chipArray}
-                title={this.getTranslation('drag-and-drop-tips')}
-                onDragEnter={this.handleDragEnter}
-                onDragOver={this.preventAndStop}
-                onDragLeave={this.handleDragLeave}
-                onDrop={this.handleFileDrop}
-            />,
-        ];
+                onFilesSelected={this.handleNewFiles}
+            >
+                <limel-file-input
+                    accept={this.accept}
+                    disabled={this.disabled || this.readonly || !!this.value}
+                >
```

Comment:
In one iteration of this PR I wrapped the `<limel-chip-set>` inside a `<limel-file-input>` only if we expected it to be clickable (no file selected). I changed it into this but just now realised we need this change https://github.com/Lundalogik/lime-elements/pull/2820/commits/a85e952b0f09a415ea4e25071db0f5ff2e371a02 to make it work.

It's maybe a bit strange to have it like this if we would like to make the component work with multiple files? Maybe that's a headache for another time ... ?

This comment is also relevant in this context https://github.com/Lundalogik/lime-elements/pull/2820/files#r1521124280

Reply:
> Maybe that's a headache for another time ... ?

Yes, for another time :)